framework/src/base/ComputeJacobianThread.C

```diff
@@ -222,6 +239,47 @@ ComputeJacobianThread::onInternalSide(const Elem *elem, unsigned int side)
 }
 
 void
+ComputeJacobianThread::onInterface(const Elem *elem, unsigned int side, BoundaryID bnd_id)
+{
+  std::vector<DGKernel *> interface_dgs;
+  _sys.getDGKernelWarehouse(_tid).activeInterfaceDG(bnd_id, interface_dgs);
+
+  if (interface_dgs.size() == 0)
+    return;
+
+  // Pointer to the neighbor we are currently working on.
+  const Elem * neighbor = elem->neighbor(side);
+
+  // Get the global id of the element and the neighbor
+  const dof_id_type
+    elem_id = elem->id(),
+    neighbor_id = neighbor->id();
+
+  if ((neighbor->active() && (neighbor->level() == elem->level()) && (elem_id < neighbor_id)) || (neighbor->level() < elem->level()))
+  {
+    _fe_problem.reinitNeighbor(elem, side, _tid);
+
+    _fe_problem.reinitMaterialsFace(elem->subdomain_id(), _tid);
+    _fe_problem.reinitMaterialsNeighbor(neighbor->subdomain_id(), _tid);
+
+    // Set the active boundary id so that BoundaryRestrictable::_boundary_id is correct
+    _fe_problem.setCurrentBoundaryID(bnd_id);
```

Comment:
After some investigation, I don't really see why I need to do this. I modeled this after https://github.com/idaholab/moose/blob/devel/framework/src/base/ComputeJacobianThread.C#L171 but I can't see it's importance here (it's not immediately obvious to me that it's necessary in the linked line either but I just may not have dug deep enough)